topostats/grains.py

```diff
@@ -158,23 +158,40 @@ def remove_noise(self, image: np.ndarray) -> np.ndarray:
         Returns
         -------
         np.ndarray
-            2D Numpy array of image with objects < absolute_smallest_grain_size removed.
+            2D Numpy array of image with objects < smallest_grain_size_nm2 removed.
         """
-        LOGGER.info(f"[{self.filename}] : Removing noise (< {self.absolute_smallest_grain_size})")
-        return remove_small_objects(image, min_size=self.absolute_smallest_grain_size)
+        LOGGER.info(
+            f"[{self.filename}] : Removing noise (< {self.smallest_grain_size_nm2} nm^2"
+            "{self.smallest_grain_size_nm2 / (self.pixel_to_nm_scaling**2):.2f} px^2)"
+        )
+        return remove_small_objects(
+            image, min_size=self.smallest_grain_size_nm2 / (self.pixel_to_nm_scaling**2), **kwargs
+        )
 
     def remove_small_objects(self, image: np.array, **kwargs):
```

Comment:
I hadn't noticed this before but there is potential here for conflict since we have imported `remove_small_objects` from skimage and this redefines that name, albeit as a method to the `Grains` class.

I noticed it reading the function and thinking "_Why is this function using recursion?_" then checking what had been imported.

Sorry to be a pain but I think we could perhaps avoid this by instead...

```
from skimage import morphology
```

and then calling `morphology.remove_small_objects` on line 187 and on line 128 using `morphology.label()`. This is the suggestion I read about in [this article](http://blog.nicholdav.info/four-tips-structuring-research-python/)

Reply:
I agree! I noticed this while writing this PR and forgot to address it.